drivers/storage/ide/atapi/dev_identify.c

```diff
@@ -0,0 +1,359 @@
+/*
+ * PROJECT:     ReactOS ATA Port Driver
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Device identification
+ * COPYRIGHT:   Copyright 2024-2025 Dmitry Borisov <di.sean@protonmail.com>
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include "atapi.h"
+
+/* FUNCTIONS ******************************************************************/
+
+static
+BOOLEAN
+AtaDeviceIdentifyDataEqual(
+    _In_ PIDENTIFY_DEVICE_DATA IdentifyData1,
+    _In_ PIDENTIFY_DEVICE_DATA IdentifyData2)
+{
+    if (!RtlEqualMemory(IdentifyData1->SerialNumber,
+                        IdentifyData2->SerialNumber,
+                        RTL_FIELD_SIZE(IDENTIFY_DEVICE_DATA, SerialNumber)))
+    {
+        return FALSE;
+    }
+
+    if (!RtlEqualMemory(IdentifyData1->FirmwareRevision,
+                        IdentifyData2->FirmwareRevision,
+                        RTL_FIELD_SIZE(IDENTIFY_DEVICE_DATA, FirmwareRevision) +
+                        RTL_FIELD_SIZE(IDENTIFY_DEVICE_DATA, ModelNumber)))
```

Comment:
Comparing 2 fields in one go can lead to static analysis errors. Might be worth to either use a structure that wraps `FirmwareRevision` and `ModelNumber` or do 2 seperate calls to `RtlEqualMemory`.

Reply:
Ok, fixed.